allennlp/training/trainer.py

```diff
@@ -512,14 +508,16 @@ def _validation_loss(self) -> Tuple[float, int]:
         if self._moving_average is not None:
             self._moving_average.assign_average_value()
 
-        if self._validation_iterator is not None:
-            val_iterator = self._validation_iterator
+        if self._validation_data_loader is not None:
+            validation_data_loader = self._validation_data_loader
         else:
-            val_iterator = self.iterator
+            raise ConfigurationError(
+                "Validation results cannot be calculated without a validation_data_loader"
+            )
 
-        val_generator = val_iterator(self._validation_data, num_epochs=1, shuffle=False)
-        num_validation_batches = val_iterator.get_num_batches(self._validation_data)
-        val_generator_tqdm = Tqdm.tqdm(val_generator, total=num_validation_batches)
+        val_generator_tqdm = Tqdm.tqdm(
+            iter(validation_data_loader), total=len(validation_data_loader)
```

Comment:
Does it not work to just say `Tqdm.tqdm(validation_data_loader)`, and then it will also work if `__len__` is not available?

Reply:
oh yep, nice 👍